### PR TITLE
[web][battery] Fix battery export bug on web

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

importing battery breaks on web

# How

ensure module.exports exists like we did in font, FS, etc...

### CODE REVIEW

File: packages/expo-battery/src/Battery.ts [hunk 1]
```diff
@@ -18,19 +18,21 @@ const BatteryEventEmitter = new EventEmitter(ExpoBattery);
...
-  isSupported: {
-    enumerable: true,
-    get() {
-      deprecate('expo-battery', 'Battery.isSupported', {
-        replacement: 'Battery.isAvailableAsync',
-        currentVersion: require('../package.json').version,
-        versionToRemove: '3.0.0',
-      });
-      return (ExpoBattery && ExpoBattery.isSupported) || false;
+if (module.exports) {
```
Comment (sjchmiela) [inline packages/expo-battery/src/Battery.ts]: The only place I found `if (module.exports)` was 

https://github.com/expo/expo/blob/8076e32814eba2b07729c2b724d92186fc0ff206/packages/expo/src/Expo.fx.tsx#L50

Should we do exactly the same here?

Comment (tsapeta) [reply packages/expo-battery/src/Battery.ts]: imho `module` may not be defined in some contexts and if we use JS strict mode then we should do:

`if (global.module && module.exports)`

without `global.` we could get `ReferenceError: module is not defined`

Comment (sjchmiela) [reply packages/expo-battery/src/Battery.ts]: fwiw prism uses just `module` to do the same thing
https://github.com/expo/expo/blob/98b9551b9eaf2258ac397362042244269ee09f3c/docs/static/libs/prism/prism.js#L332

Comment (tsapeta) [reply packages/expo-battery/src/Battery.ts]: this is just another way to prevent reference error I mentioned 😛

Comment (SimenB) [reply packages/expo-battery/src/Battery.ts]: Note that in node `module` is not a property of `global` (it's injected into each module's scope as as argument), so a `typeof` check is a better idea

Comment (tsapeta) [reply packages/expo-battery/src/Battery.ts]: that's fair point @SimenB, thanks 😉

**Pull request merged**